### PR TITLE
sqlite on scaffold only

### PR DESCRIPTION
**Pull request by iamshanvy**
use sqlite prisma provider only when scaffolding the project, use postgresql by default (for demo deployment)

essentially it is just a string replacement from postgresql to sqlite in `schema.prisma` file

alternative is probably having a separate file (like `scaffold.schema.prisma`) that would be `mv`ed to scaffolded project with provider set as `sqlite`, main idea is to use sqlite only on scaffolded project and default to postgres in other cases

proposed solution for https://github.com/RubricLab/create-rubric-app/issues/16, open for discussion

marked as draft since I am able to test on windows only tomorrow

### CODE REVIEW

**Comment by tedspare** [comment]
super creative approach! thanks for doing this work.

I wonder if projects scaffolded with `create-rubric-app` might still want to use `postgres` when deployed

another approach is having two schema files, `dev.prisma.schema` and `prisma.schema`, then pointing to the right file depending by env eg.

```bash
prisma generate --schema=$(if [ $NODE_ENV = 'production' ]; then echo schema.prisma; else echo dev.schema.prisma; fi)
```

would you be up to trying that approach? (this work would not be lost. iterative process)

**Comment by iamshanvy** [comment]
> super creative approach! thanks for doing this work.
> 
> I wonder if projects scaffolded with `create-rubric-app` might still want to use `postgres` when deployed
> 
> another approach is having two schema files, `dev.prisma.schema` and `prisma.schema`, then pointing to the right file depending by env eg.
> 
> ```shell
> prisma generate --schema=$(if [ $NODE_ENV = 'production' ]; then echo schema.prisma; else echo dev.schema.prisma; fi)
> ```
> 
> would you be up to trying that approach? (this work would not be lost. iterative process)

you are right, after a fresh look today I think current PR is too complex for no reason, I got pigeon holed, working on changes now

**Comment by iamshanvy** [comment]
@tedspare  the script does the job, but if windows support is needed then it would more complicated

I propose two solutions, have a simple db.js file to manage prisma related commands (since js code is cross platform here as opposed to bash) and as an example then run from package.json as `node db.js`

or

simply have separate scripts for dev and prod, like in package.json have it as `"db:push:dev":"..."` or `"db:push:prod":"..."`

**Comment by tedspare** [comment]
> @tedspare the script does the job, but if windows support is needed then it would more complicated
> 
> I propose two solutions, have a simple db.js file to manage prisma related commands (since js code is cross platform here as opposed to bash) and as an example then run from package.json as `node db.js`
> 
> or
> 
> simply have separate scripts for dev and prod, like in package.json have it as `"db:push:dev":"..."` or `"db:push:prod":"..."`
> 
> this is already assuming that we have two prisma schemas, one for dev and for prod

the latter solution sounds clean! the `postinstall` script would also be replaced with env-specific steps eg. `"build": "db:push:prod && ..."` vs `"dev": "db:push:dev && ..."`

want to give that a try?

File: templates/agent/package.json [hunk 1]
```diff
@@ -38,6 +38,8 @@
 	"private": true,
 	"scripts": {
 		"build": "next build",
+		"db:dev:push": "prisma db push --schema ./prisma/dev.schema.prisma && prisma db seed --schema ./prisma/dev.schema.prisma",
```
Comment (tedspare) [inline templates/agent/package.json:41]: A few things from testing locally:
- Change `--schema ./path` to `--schema=./path`
- Change `dev.schema.prisma` to `dev/schema.prisma`
- Remove `db seed` command altogether (I couldn't get it to work with our TSConfig)

File: templates/agent/prisma/dev.schema.prisma [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// This is your Prisma schema file,
```
Comment (tedspare) [inline templates/agent/prisma/dev.schema.prisma:1]: See [above](https://github.com/RubricLab/create-rubric-app/pull/21/files#r1403766540); let's move this to **prisma/dev/schema.prisma**

File: templates/agent/package.json
Comment (tedspare) [inline templates/agent/package.json:1]: Also, let's bump `langchain` to `"^0.0.197-rc.1"` (can simply run `bun rm langchain && bun add langchain`)

**Review by tedspare — CHANGES_REQUESTED** [review]
Looking clean! Added a few comments based on testing. Will test again once implemented.

**Review by tedspare — APPROVED** [review]
LGTM

**Comment by DexterStorey** [comment]
kick ass work @iamshanvy - been keeping tabs on this one from the shadow realm. Excited to see this on main!

**Pull request merged**